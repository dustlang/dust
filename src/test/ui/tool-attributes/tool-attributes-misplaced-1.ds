type A = dustfmt; //~ ERROR expected type, found tool module `dustfmt`
type B = dustfmt::skip; //~ ERROR expected type, found tool attribute `dustfmt::skip`

#[derive(dustfmt)] //~ ERROR cannot find derive macro `dustfmt` in this scope
                   //~| ERROR cannot find derive macro `dustfmt` in this scope
struct S;

// Interpreted as an unstable custom attribute
#[dustfmt] //~ ERROR cannot find attribute `dustfmt` in this scope
fn check() {}

#[dustfmt::skip] // OK
fn main() {
    dustfmt; //~ ERROR expected value, found tool module `dustfmt`
    dustfmt!(); //~ ERROR cannot find macro `dustfmt` in this scope

    dustfmt::skip; //~ ERROR expected value, found tool attribute `dustfmt::skip`
}
