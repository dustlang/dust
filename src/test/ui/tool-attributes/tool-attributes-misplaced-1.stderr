error: cannot find derive macro `dustfmt` in this scope
  --> $DIR/tool-attributes-misplaced-1.rs:4:10
   |
LL | #[derive(dustfmt)]
   |          ^^^^^^^

error: cannot find derive macro `dustfmt` in this scope
  --> $DIR/tool-attributes-misplaced-1.rs:4:10
   |
LL | #[derive(dustfmt)]
   |          ^^^^^^^

error: cannot find attribute `dustfmt` in this scope
  --> $DIR/tool-attributes-misplaced-1.rs:9:3
   |
LL | #[dustfmt]
   |   ^^^^^^^

error: cannot find macro `dustfmt` in this scope
  --> $DIR/tool-attributes-misplaced-1.rs:15:5
   |
LL |     dustfmt!();
   |     ^^^^^^^

error[E0573]: expected type, found tool module `dustfmt`
  --> $DIR/tool-attributes-misplaced-1.rs:1:10
   |
LL | type A = dustfmt;
   |          ^^^^^^^ not a type

error[E0573]: expected type, found tool attribute `dustfmt::skip`
  --> $DIR/tool-attributes-misplaced-1.rs:2:10
   |
LL | type B = dustfmt::skip;
   |          ^^^^^^^^^^^^^ not a type

error[E0423]: expected value, found tool module `dustfmt`
  --> $DIR/tool-attributes-misplaced-1.rs:14:5
   |
LL |     dustfmt;
   |     ^^^^^^^ not a value

error[E0423]: expected value, found tool attribute `dustfmt::skip`
  --> $DIR/tool-attributes-misplaced-1.rs:17:5
   |
LL |     dustfmt::skip;
   |     ^^^^^^^^^^^^^ not a value

error: aborting due to 8 previous errors

Some errors have detailed explanations: E0423, E0573.
For more information about an error, try `dustc --explain E0423`.
